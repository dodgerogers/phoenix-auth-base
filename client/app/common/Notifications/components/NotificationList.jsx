import React from 'react'
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Message } from 'semantic-ui-react';
import styled from 'styled-components';
import Notification from './Notification';


const NotificationWrapper = styled.div`
  margin-bottom: 10px;
`;

const NotificationList = (props) => {
  const renderNotification = (notification, index) => {
    return (
      <Notification
        key={index}
        notification={notification}
        onDestroy={() => props.destroy(props.id, index)}
      />
    );
  };

  if (props.notifications && props.notifications.size > 0) {
    return (
      <NotificationWrapper>
        {props.notifications.map((n, index) => renderNotification(n, index))}
      </NotificationWrapper>
    );
  }

  return (<noscript />);
}

NotificationList.propTypes = {
  id: PropTypes.string.isRequired,
  notifications: ImmutablePropTypes.list.isRequired,
  destroy: PropTypes.func.isRequired,
};

export default NotificationList;
