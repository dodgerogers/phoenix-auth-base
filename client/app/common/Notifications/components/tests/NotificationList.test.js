import React from 'react';
import { shallow, mount } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import NotificationList from '../NotificationList';


describe('NotificationList', () => {
  describe('PureComponent', () => {
    const fullProps = () => ({
      id: 'areaID',
      notifications: fromJS([
        fromJS({
          message: 'Information',
        }),
      ]),
    });

    const pureComponent = (props) => mount(
      <NotificationList {...props} />
    );

    it('matches snapshot when no notifications present', () => {
      const props = fullProps();
      props.notifications = fromJS([]);
      const component = pureComponent(props);

      expect(toJson(component)).toMatchSnapshot();
    });

    it('matches snapshot with at least one notification', () => {
      const props = fullProps();
      const component = pureComponent(fullProps());

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
