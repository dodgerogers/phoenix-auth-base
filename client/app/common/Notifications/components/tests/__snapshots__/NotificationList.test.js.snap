// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationList PureComponent matches snapshot when no notifications present 1`] = `<noscript />`;

exports[`NotificationList PureComponent matches snapshot with at least one notification 1`] = `
<styled.div>
  <Connect(Notification)
    key="0"
    notification={
      Immutable.Map {
        message: "Information",
      }
    }
    onDestroy={[Function]}
  />
</styled.div>
`;
